### ISSUE TITLE
Error ab_logisticscenter_0.18.9

### ISSUE THREAD

**Issue by kots01** [ABackerNINI/factorio_mod#2]
![Скриншот 23-06-2020 16 56 06](https://user-images.githubusercontent.com/66099307/85413911-19afb200-b574-11ea-8620-6a9704cbd17d.png) All PY mods and 'ab-logistics-center-rc-logistic-slots-count' = 12. Yes, the recommended value is not more than 5, but with PY mods it will not work without it. The mod worked stably for about 70 hours. Thanks for the cool mod, and I'm sorry if something is not clear I write with the help of a translator

**Comment by ABackerNINI** [comment]
This will happen when you put more than 500 kinds of items into the center. You can set 'ab-logistics-center-lc-item-slot-count' to 2000, default is 500.

**Comment by kots01** [comment]
Thanks so much for the quick reply.

**Comment by kots01** [comment]
Another question in PY mods is about 1000 names of plants, ores, assemblers and other items. Regulate restrictions on reception and storage in the logistics center of each item for quite some time. Is it possible in the future to make automatic settings for the limit of items that will be stored in the logistics center? Set up by subgroups. For example, assemblers and other industrial buildings = 10, ores = 1000-5000, metals 2000-5000 and so on.

**Comment by ABackerNINI** [comment]
Yes, it sounds possible and it's a pretty good idea. ^_^

**Comment by kots01** [comment]
And one more question. Is it possible to add several types of warehouses with large capacities, for example like these in 150 and 800?
![Скриншот 27-06-2020 16 09 13](https://user-images.githubusercontent.com/66099307/85923133-38be8480-b891-11ea-8f0d-660fe4bc0c77.png)

**Comment by ABackerNINI** [comment]
Do you mean 3*3 or 6*6 requester/collector chests with large capacity of 150/800?

**Comment by kots01** [comment]
![Скриншот 27-06-2020 19 24 54](https://user-images.githubusercontent.com/66099307/85927143-e8552000-b8ac-11ea-9651-46ac9dd0cc81.png)When using rail transportation, a large warehouse with 800 capacity is used as a request. From it it is possible to send 6-12 or 18  transport belts and it is convenient for PY mods. Medium warehouse 150 capacity is mainly used as a provider. I could be wrong, but similar in size warehouses are in the mods of bob and angel. There is also a proposal to make barrels for teleporting liquids into them, the vanilla value of 50 for one barrel seems small enough, maybe 500-1000 will be just right. Barrel and storage recipes can be made more expensive.

**Comment by ABackerNINI** [comment]
It's possible but there is one problem, it's hard to find the proper images for those 3*3/6*6 size chests, since I'm not good at PS. 
Please send me if you find ones, thanks.

**Comment by kots01** [comment]
![logistics-center](https://user-images.githubusercontent.com/66099307/86013358-bc11de80-ba27-11ea-905e-398fa92a11f6.png)
![lab-mk3](https://user-images.githubusercontent.com/66099307/86013527-ea8fb980-ba27-11ea-8efe-8569b376943e.png)
![logistics-center](https://user-images.githubusercontent.com/66099307/86013763-2f1b5500-ba28-11ea-96b6-c6316ff76240.png)
The first two pictures are made by me, the third picture for comparison is taken from your mod. If you liked one of them, I can do something similar either in different colors as 1 or in the same color as 2. As for the warehouse, it will be more difficult, I have no such experience yet. I'll see if I can find something.

**Comment by ABackerNINI** [comment]
They look very nice! Actually all images of this mod are copied from other mods (I have declared that in the mod). I'll really appreciate if you'd like to make them look different. I want to try the first one. ლ(⌒▽⌒ლ)

There are several requirements of the picture: The pic should be png format, transparent. Size of each one is 113*91, it's OK if there are slightly different, but keep them the same. The shadow.

It shouldn't need to change the code if you keep their size as 113*91, just unzip the mod, replace the pic and start the game.
Edit the code if you change their size, _/prototypes/logistics-center.lua_ line 6/7.

Thanks, ٩(◕‿◕｡)۶

**Comment by kots01** [comment]
Yes, now I looked at the values ​​of the picture of the laboratory 98 * 87.  I think it will be easier for me to edit two lines of your code and leave the vanilla sizes. Is it possible to increase the number of rows in the picture? (3 * 11 vanilla) for example to make 6 * 11 or 3 * 22? Thus add a palette of shades. frame_count = 33, line_length = 11, animation_speed = 1 / 3?. Another question. There is an LTN mod and it has an interesting parameter, request priority and supply priority, if I have two or more requests of one item, it becomes possible to choose which request will be served with high priority. Sometimes this can be useful when distributing expensive items, but I don’t know how much this can complicate the teleportation mode.

**Comment by ABackerNINI** [comment]
Emmmm, you also need to change the code _logistics-center-animation.lua_ line 22/23/25/26. Any number of rows should be ok. frame_count means the total number of small pics. line_length means columns of each row in the picture.

It seems difficult to implement request/supply priority, but it's possible. I'll think about it later.

**Comment by kots01** [comment]
Ready preliminary version of the pictures. 

[ab_logisticscenter_0.18.9.zip](https://github.com/ABackerNINI/factorio_mod/files/4857223/ab_logisticscenter_0.18.9.zip)

 Made adjustments in the code according to your recommendations, and also slowed down the animation speed to 1/9. The picture now has 9 * 11 instead of 3 * 11. There are rows with smooth transitions of shades and there are sharp transitions immediately to a different color. The last two rows are vanilla for comparison. Choose which style you like best, which shades it is better to leave, and which to remove. 

Tell me, is it possible without damaging saves to increase the size of the logistics center? Maybe this way the animation will be more noticeable.

**Comment by ABackerNINI** [comment]
* It looks really nice! There is one small problem, it moves slightly, check the following video.

* [Factorio 0.18.34 record.zip](https://github.com/ABackerNINI/factorio_mod/files/4863970/Factorio.0.18.34.2020-07-02.19-34-35.zip)

* Size of the logistics center? Do you mean selection and collision box size?  _/prototypes/logistics-center.lua_ line 6/7 defines its selection and collision box size. And changing it should not damage the save, but may need a little adjustments of the code.

* ![Screen shot](https://user-images.githubusercontent.com/12003373/86356397-c4c01a00-bc9e-11ea-8175-e255655be211.png)

**Comment by kots01** [comment]
If you are talking about animation speed, I reduced it from 1/5 to 1/9 to see the difference. It can be returned as it was

Yes, if you can increase the size of the logistics center as in the picture on the left (or maybe a little more).

I will try to add a few more rows of shades for a change.

**Comment by ABackerNINI** [comment]
I mean the logistics center moves several pixels to the left and then moves back after several seconds.

**Comment by kots01** [comment]
Yes, now I noticed. The reason is found, it will be fixed

**Comment by kots01** [comment]
I corrected the movement of the logistics center, but found another problem. When adding additional rows, the picture is shifted a few pixels to the side. Perhaps you know how to fix this? When adding rows, the proportions were observed. 3 * 11 = 1078 * 261, 9 * 11 = 1078 * 783.
![Скриншот 03-07-2020 12 57 12](https://user-images.githubusercontent.com/66099307/86479337-a0bf1000-bd54-11ea-81e6-c1529e30ee4b.png)

**Comment by ABackerNINI** [comment]
This is not a big problem, I can fix it.

If you want, adjust the code _logistics-center-animation.lua_ line 28 and _logistics-center.lua_ line 31, shift = {X,Y}.

**Comment by kots01** [comment]
See which option you like more this or the previous one. I also slightly changed the color of the charging of the logistics center and chests (the reaction of the mod authors to this is unknown). If possible, send an animation video with an enlarged logistics center. Maybe some colors are too bright?
[ab_logisticscenter_0.18.9.zip](https://github.com/ABackerNINI/factorio_mod/files/4881358/ab_logisticscenter_0.18.9.zip)

**Comment by ABackerNINI** [comment]
* It's really really nice! Awesome!

* Make a pull request and upload these pictures, so that you'll become a contributor to this mod. I'll do the rest coding part.

* Thank you!

* [Captures.zip](https://github.com/ABackerNINI/factorio_mod/raw/master/Captures.zip)

* Please tell me if you can't download the video.

**Comment by kots01** [comment]
A small bug was found in the energy bar animation. This will be fixed.

**Comment by kots01** [comment]
I understand you correctly, I need to open a pull request to select lc_0.16.0 and upload photo animations? I am just learning)

**Comment by ABackerNINI** [comment]
* It's quite simple, but you need to commit to master branch, not lc_0.16.0.

**Comment by kots01** [comment]
I searched the Internet a bit and found some images of the warehouse. It looks like they are in the public domain, if you like something, I will try to replace the colors with the ones suitable for the mod. But there are a few questions I don’t understand to what size they need to be compressed, and what picture quality will be in the game.
![request](https://user-images.githubusercontent.com/66099307/87716922-42644980-c7b8-11ea-864c-425ac5194582.jpg)
![collector](https://user-images.githubusercontent.com/66099307/87716933-44c6a380-c7b8-11ea-9842-59053ba813ad.jpg)
![2](https://user-images.githubusercontent.com/66099307/87716944-4a23ee00-c7b8-11ea-8e66-6fbbb743261d.png)
![3](https://user-images.githubusercontent.com/66099307/87716953-4ee8a200-c7b8-11ea-9388-0a8f418a58bb.png)
![4](https://user-images.githubusercontent.com/66099307/87716967-53ad5600-c7b8-11ea-963b-832c40921c42.png)
![5](https://user-images.githubusercontent.com/66099307/87716969-5445ec80-c7b8-11ea-9fb0-2d2778ef0b78.jpg)
